src/ControlSystem/UpdateFunctionOfTime.hpp

```diff
@@ -58,4 +76,174 @@ struct ResetFunctionOfTimeExpirationTime {
+ * should be one `UpdateAggregator` for every group of control systems that have
+ * the same `control_system::protocols::Measurement`.
+ */
+struct UpdateAggregator {
+  UpdateAggregator() = default;
+
+  /*!
+   * \brief Construct a new UpdateAggregator using a set of control system
+   * names.
+   *
+   * It is expected that all the control systems in this set use the same
+   * `control_system::protocols::Measurement`.
+   */
+  UpdateAggregator(std::unordered_set<std::string> control_system_names);
+
+  /*!
+   * \brief Inserts and stores information for one of the control systems that
+   * this class was constructed with.
+   *
+   * \param control_system_name Name of control system to add information for
+   * \param new_measurement_timescale DataVector of new measurement timescales
+   * calculated from a control system update.
+   * \param new_measurement_expiration_time New measurement expiration time
+   * calculated during that update.
+   * \param control_signal New highest derivative for the function of time
+   * calculated during that update (will be `std::move`ed).
+   * \param new_fot_expiration_time New function of time expiration time
+   * calculated for during that update
+   */
+  void insert(const std::string& control_system_name,
+              const DataVector& new_measurement_timescale,
+              const double new_measurement_expiration_time,
+              DataVector&& control_signal,
+              const double new_fot_expiration_time);
+
+  /*!
+   * \brief Checks if `insert` has been called for all control systems that this
+   * class was constructed with.
+   */
+  bool is_ready() const;
+
+  /*!
+   * \brief Returns a sorted concatenation of the control system names this
+   * class was constructed with.
+   *
+   * The sorting is done with `std::sort`.
+   */
+  const std::string& combined_name() const;
+
+  /*!
+   * \brief Once `is_ready` is true, returns a map between the control system
+   * name and a `std::pair` containing the `control_signal` that was passed to
+   * `insert` and the minimum of all the `new_fot_expiration_time`s passed to
+   * `insert`, respectively.
+   *
+   * \details This function is expected to only be called when `is_ready` is
+   * true, and only be called once. It also must be called before
+   * `combined_measurement_expiration_time`. This function leaves all stored
+   * data in tact.
+   */
+  std::unordered_map<std::string, std::pair<DataVector, double>>
+  combined_fot_expiration_times() const;
+
+  /*!
+   * \brief Once `is_ready` is true, returns a `std::pair` containing the
+   * minimum of all `new_measurement_timescale`s passed to `insert` and the
+   * minimum of all `new_measurement_expiration_time`s passed to `insert`,
+   * respectively.
+   *
+   * \details This function is expected to only be called when `is_ready` is
+   * true, and only be called once. It also must be called after
```

Comment:
"once per something" or something like that.  (I assume you don't create an entire new UpdateAggregator for every update.)